src/ucp/api/ucp.h

```diff
@@ -333,6 +335,19 @@ typedef enum {
 
 /**
  * @ingroup UCP_DATATYPE
+ * @brief Generate a reusable identifier for Scatter-gather IOV data type.
```

Comment:
1. Let's say you have an IOV of length 10. If "is_reusable" is on - then the first time you create 11 mkeys (one for each mkey + one for the UMR), and second time you just use the existing (UMR) key to send'em. If "is_reusable" is off - first time you register the mkeys (only 10 are stored in the rcache, because the UMR can't be indexed by address only), and every time you send that IOV you do 10 lookups. These 10 lookups are avoided with "is_reusable". Why not have "is_reusable" all the time? 
A. Backward-compatibility - the API currently does not require free-ing datatypes, "is_reusable" does.
B. "is_reusable" is a more complex API compared to a simple const datatype indicator. Do we want to disallow the simpler API?

2. if you're asking "can we use a struct from the user rather than a handle?" - then yes, we can extend `struct ucp_dt_iov` to include a "memh-cache" field to be filled by UCX - but then the user would have to destroy it eventually. if you're asking about a new "struct datatype" - if i understand you correctly that would just be a different name for "IOV"+"is_reusable"... are you refering to a new datatype?